### PR TITLE
Enable AddSqlServer only if chain to AddSignalR

### PR DESCRIPTION
**Pull request by Xiaohongt**
This is related to https://github.com/aspnet/SignalR-Server/pull/65:

e.g. In Startup, enable AddSqlServer only when chain to AddSignalR:

``` csharp
            app.UseServices(services =>
            {
                services.AddSignalR(options =>
                        {
                            options.Hubs.EnableDetailedErrors = true;
                        })
                        .AddSqlServer(options =>
                        {
                            options.ConnectionString = "<connection string>";
                        });
            });
```

### CODE REVIEW

File: src/Microsoft.AspNet.SignalR.SqlServer/SqlServerServicesBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+using System;
+using Microsoft.AspNet.SignalR;
+using Microsoft.AspNet.SignalR.Messaging;
+using Microsoft.AspNet.SignalR.SqlServer;
+using Microsoft.Framework.ConfigurationModel;
+
+namespace Microsoft.Framework.DependencyInjection
+{
+    public static class SignalRServiceCollectionExtensions
```
Comment (halter73) [inline src/Microsoft.AspNet.SignalR.SqlServer/SqlServerServicesBuilderExtensions.cs]: This doesn't match the file name. Just make it `SqlServerSignalRServicesBuilderExtensions`. Good thing no one should ever have to actually write the class name :wink:.

Comment (halter73) [inline src/Microsoft.AspNet.SignalR.SqlServer/SqlServerServicesBuilderExtensions.cs]: You can :ship: after you change the class and file name.

File: src/Microsoft.AspNet.SignalR.SqlServer/SqlServerServicesBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        public static SignalRServicesBuilder AddSqlServer(this SignalRServicesBuilder builder, Action<SqlScaleoutConfiguration> configureOptions = null)
+        {
+            return builder.AddSqlServer(configuration: null, configureOptions: configureOptions);
+        }
+        public static SignalRServicesBuilder AddSqlServer(this SignalRServicesBuilder builder, IConfiguration configuration, Action<SqlScaleoutConfiguration> configureOptions = null)
+        {
+            builder.ServiceCollection.AddSingleton<IMessageBus, SqlMessageBus>();
```
Comment (halter73) [inline src/Microsoft.AspNet.SignalR.SqlServer/SqlServerServicesBuilderExtensions.cs]: Before this method was using a `ServiceDescriber`. I think we should continue using that.

**Comment by BrennanConroy** [comment]
:shipit:

**Pull request closed without merging**